src/ophys_etl/pipelines/brain_observatory/scripts/run_mesoscope_splitting.py

```diff
@@ -58,12 +60,15 @@ def convert_column(input_tif, session_storage, experiment_info, **h5_opts):
     return conversion_output(mt.volume_views[0], filename, experiment_info)
 
 
-def split_z(input_tif, experiment_info, **h5_opts):
+def split_z(input_tif, experiment_info, testing=False, **h5_opts):
```

Comment:
This is not a good practice. You are making a path for you function to follow when testing which is different from when it is actuall being used. Let's come up with a better solution.

Reply:
Agree. You should probably be using using mock `patch` or pytest `monkeypatch` fixture and replacing the `MesoscopeTiff` call instead of having a `testing=False` parameter in your function.

Reply:
As Nick surmised: I was trying to get `split_z` into a state where it could accept a class other than `MesoscopeTiff`, which requires a full scanimage TIFF file which is, apparently, nigh impossible to generate by hand (at least, I could find no good tools for doing so). The unit test that calls this function implements a dummy class `MesoscopeTiffDummy` that has all of the salient features of `MesoscopeTiff` without needing the weighty scanimage meatadata header.

Would altering `split_z` to accept a kwarg `tiff_reader_class=MesoscopeTiff` (which would be redirected to the dummy class in the unit test) be a reasonable solution?

Reply:
If you look at how I implemented my tests, I basically just subclassed the MesoscopeTiff class and overwrote the fields that I needed in the `__init__`

Reply:
> Would altering split_z to accept a kwarg tiff_reader_class=MesoscopeTiff (which would be redirected to the dummy class in the unit test) be a reasonable solution?

not really, that is effectively the same bypassing of the production code.

Reply:
Then should we change the call signature of `split_z` in

`/src/ophys_etl/pipeliens/brain_observatory/scripts/run_mesoscope_splitting.py`

?

`split_timeseries` and `split_image` both accept `MesoscopeTiff` objects as their first argument. `split_z` accepts a path to a TIFF file as its first argument, which it immediately uses to instantiate a `MesoscopeTiff`. I'm not sure there's a very good reason these three functions should have different call signatures.

Reply:
I actually changed them to accept `MesoscopeTiff` objects, they originally all took strings. Sorry I guess I didn't mention that...

Reply:
Done. I consolidated the z_split tests into the testing framework that I had already setup for my other tests.